packages/components/pie-input/test/component/pie-input.spec.ts

```diff
@@ -796,6 +796,98 @@ test.describe('PieInput - Component tests', () => {
                 expect(inputShell).toHaveAttribute('size', 'large');
             });
         });
+
+        test.describe('required', () => {
+            test('should not render a required attribute on the input element if no required provided', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieInput, {});
+
+                // Act
+                const input = component.locator('input');
+
+                // Assert
+                expect((await input.getAttribute('required'))).toBe(null);
+            });
+
+            test('should apply the required prop to the HTML input rendered', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieInput, {
+                    props: {
+                        required: true,
+                    } as InputProps,
+                });
+
+                // Act
+                const input = component.locator('input');
+
+                // Assert
+                expect((await input.getAttribute('required'))).toBe('');
```

Comment:
i might be missing something, but based on [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute), `getAttribute` will return `null `or empty string when the attribute isn't found but the test is a positive clause. 

maybe we can use [hasAttribute](https://developer.mozilla.org/en-US/docs/Web/API/Element/hasAttribute)

Reply:
Yeah happy to change

Reply:
<img width="859" alt="Screenshot 2024-04-09 at 14 54 10" src="https://github.com/justeattakeaway/pie/assets/16766185/1b3b768f-3473-4493-a450-e113778a5811">

I don't think our version of playwright supports that method!

Reply:
are we sure that the empty string here means the `required` attribute is found not `doesn't exist` as per the MDN spec?

Reply:
According to the spec you linked:

"All modern web browsers return null when the specified attribute does not exist on the specified element."

And we can see this here:
<img width="353" alt="Screenshot 2024-04-09 at 14 51 19" src="https://github.com/justeattakeaway/pie/assets/16766185/635503d6-d43e-44ac-a9a8-4d17f79500a3">

I am unsure how else to test this but I am fully open to suggestions

Reply:
well the browser is a source of truth and it's returning an empty string when it's assigned LOL. I would say we should use the hasAttrubte but seems it's missing. 🤝

Reply:
I fully agree 😃 hopefully when we update playwright we can revisit!